apps/studio/components/interfaces/APIKeys/APIKeyRow.tsx

```diff
@@ -0,0 +1,193 @@
+    error,
+    refetch: refetchApiKey,
+  } = useAPIKeyIdQuery(
+    {
+      projectRef,
+      id: apiKey.id as string,
+      reveal: true,
+    },
+    {
+      enabled: show,
+      staleTime: 0, // Data is considered stale immediately
+      cacheTime: 0, // Cache is cleared immediately after query becomes stale
+    }
+  )
+
+  async function onSubmitShow() {
+    setShowState(true)
+    // Set a timeout to invalidate the cache after a certain amount of time
+    setTimeout(() => {
+      setShowState(false)
+      queryClient.removeQueries({
+        queryKey: apiKeysKeys.single(projectRef, apiKey.id as string),
+        exact: true,
+      })
+    }, 10000) // Destroy query after 10 seconds
+  }
+
+  async function onCopy() {
+    // if ID already exists from a reveal action, return that
+    if (data?.api_key) return data?.api_key
+
+    try {
+      // fetch ID and then destroy query immediately
+      const result = await refetchApiKey()
+      queryClient.removeQueries({
+        queryKey: apiKeysKeys.single(projectRef, apiKey.id as string),
+        exact: true,
+      })
+
+      if (result.isSuccess) return result.data.api_key
+
+      if (error) {
+        toast.error('Failed to copy secret API key')
+      }
+    } catch (error) {
+      console.error('Failed to fetch API key:', error)
+    }
+
+    // Fallback to the masked version
+    return apiKey.api_key
+  }
+
+  return (
+    <>
+      <div
+        className={cn(
+          InputVariants({ size: 'tiny' }),
+          'flex-1 grow gap-0 font-mono rounded-full max-w-60 overflow-hidden',
+          show ? 'ring-1 ring-foreground-lighter ring-opacity-50' : 'ring-0 ring-opacity-0',
+          'transition-all'
+        )}
+      >
+        <AnimatePresence mode="wait" initial={false}>
+          <motion.span
+            key={show ? 'shown' : 'hidden'}
+            initial={{ opacity: 0, y: show ? 16 : -16 }}
+            animate={{ opacity: 1, y: 0 }}
+            exit={{ opacity: 0, y: show ? 16 : -16 }}
+            transition={{
+              duration: 0.12,
+              y: { type: 'spring', stiffness: 2450, damping: 55 },
+            }}
+            className="truncate"
+          >
+            {show ? data?.api_key : apiKey?.api_key}
+          </motion.span>
+        </AnimatePresence>
+      </div>
+      {isSecret && (
+        <AnimatePresence>
+          {!show && (
+            <motion.div
+              initial={{ opacity: 0, scale: 1, width: 0 }}
+              animate={{ opacity: 1, scale: 1, width: 'auto' }}
+              exit={{ opacity: 0, scale: 1, width: 0 }}
+              transition={{ duration: 0.12 }}
+              style={{ overflow: 'hidden' }}
+            >
+              <Button
+                type="outline"
+                className="rounded-full px-2"
+                icon={<Eye strokeWidth={2} />}
+                onClick={onSubmitShow}
+              />
+            </motion.div>
+          )}
+        </AnimatePresence>
```

Comment:
i'm not sure what this button does - should we add a tooltip here? The UX after involves a big UI shift here too and I'm not sure whats the expected outcome

https://github.com/user-attachments/assets/7f6037f4-1ba1-4af4-8b32-ea56ea9487b8

Reply:
its a bug

Reply:
happy to merge in as is, as we're going to rewrite it